ee/tabby-webserver/src/schema/auth.rs

```diff
@@ -301,6 +301,37 @@ pub struct PasswordResetInput {
     pub password2: String,
 }
 
+#[derive(Validate, GraphQLInputObject)]
+pub struct PasswordUpdateInput {
```

Comment:
add an optional `old_password` field.

The logic would be:
1. If the user doesn't have password set (empty password, registered from oauth signin), they don't need `old_password` to change the password.
2. If user already have password set, they we need to verify the old password.

---

Other TODOs:
1. adding a field `is_password_set` to the userDAO.
2. Aadd an GraphQL query api `me` to returns `UserDAO` for the user, and